Content.Server/Physics/Controllers/MoverController.cs

```diff
@@ -95,44 +89,121 @@ private void HandleShuttleMovement(SharedPlayerInputMoverComponent mover)
                 return;
             }
 
-            // Depending whether you have "cruise" mode on (tank controls, higher speed) or "docking" mode on (strafing, lower speed)
-            // inputs will do different things.
-            // TODO: Do that
-            float speedCap;
-            var angularSpeed = 0.075f;
-
             // ShuttleSystem has already worked out the ratio so we'll just multiply it back by the mass.
-            var movement = (mover.VelocityDir.walking + mover.VelocityDir.sprinting);
+            var movement = mover.VelocityDir.walking + mover.VelocityDir.sprinting;
+            var system = EntitySystem.Get<ThrusterSystem>();
+
+            if (movement.Length.Equals(0f))
+            {
+                // TODO: This visualization doesn't work with multiple pilots so need to address that somehow.
+                system.DisableAllThrustDirections(shuttleComponent);
+                return;
+            }
+
+            var speedCap = 0f;
 
             switch (shuttleComponent.Mode)
             {
                 case ShuttleMode.Docking:
-                    if (physicsComponent.LinearVelocity.LengthSquared == 0f)
+                    system.DisableAllThrustDirections(shuttleComponent);
+                    var dockDirection = movement.ToWorldAngle().GetDir();
+                    var dockFlag = dockDirection.AsFlag();
+
+                    // Won't just do cardinal directions.
+                    foreach (DirectionFlag dir in Enum.GetValues(typeof(DirectionFlag)))
                     {
-                        movement *= 5f;
-                    }
+                        // Brain no worky but I just want cardinals
+                        switch (dir)
+                        {
+                            case DirectionFlag.South:
+                            case DirectionFlag.East:
+                            case DirectionFlag.North:
+                            case DirectionFlag.West:
+                                break;
+                            default:
+                                continue;
```

Comment:
wait do you want cardinals or not? since this breaks if its a cardinal?

Reply:
it looks weird af but the break goes out of the switch and the continue works on the loop.

Reply:
okay that is very weird but makes sense